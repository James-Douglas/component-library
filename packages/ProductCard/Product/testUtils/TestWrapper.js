import React from 'react';
import ProductProvider from '../../Context/ProductProvider';

const ProductTestWrapper = ({
  // eslint-disable-next-line react/prop-types
  children, product, onDetails = () => {}, onSelect = () => {},
}) => (
  <ProductProvider product={product} onDetails={onDetails} onSelect={onSelect}>
    { children }
  </ProductProvider>
);

export default ProductTestWrapper;
