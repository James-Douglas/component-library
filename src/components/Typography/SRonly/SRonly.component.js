import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const StyledSr = styled.div`
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  clip: rect(0, 0, 0, 0);
  white-space: nowrap;
  border-width: 0;
`;

const SRonly = ({
  children,
}) => (
  <StyledSr>
    {children}
  </StyledSr>
);

SRonly.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.string),
    PropTypes.string,
  ]),
};

SRonly.defaultProps = {
  children: '',
};

export default SRonly;
