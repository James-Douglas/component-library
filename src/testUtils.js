// test-utils.js
import {
  render, fireEvent, wait, act,
} from '@testing-library/react';
import ManorProvider from './providers/ManorProvider';

const customRender = (ui, options) => render(ui, { wrapper: ManorProvider, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export {
  customRender as render,
  // export other used functions to appease eslint
  fireEvent,
  wait,
  act,
};
