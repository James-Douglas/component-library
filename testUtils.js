import React from 'react';
import {
  render, fireEvent, wait, act,
} from '@testing-library/react';
import ManorProvider from './packages/Provider/ManorProvider';

const ManorTestWrapper = ({ children }) => (
  <ManorProvider>
    {children}
  </ManorProvider>
);

const customRender = (ui, options) => render(ui, { wrapper: ManorTestWrapper, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export {
  customRender as render,
  // export other used functions to appease eslint
  fireEvent,
  wait,
  act,
};
