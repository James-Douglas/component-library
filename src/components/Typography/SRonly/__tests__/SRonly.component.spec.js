import React from 'react';
import { render } from '../../../../testUtils';
import 'jest-styled-components';
import SRonly from '../SRonly.component';

describe('SRonly.component', () => {
  it('renders correctly with minimal props', () => {
    const { container } = render(<SRonly />);
    expect(container).toMatchSnapshot();
  });

  it('renders with its style', () => {
    const { getByText } = render(
      <SRonly>
        SRonly
      </SRonly>,
    );
    expect(getByText('SRonly')).toBeDefined();
    expect(getByText('SRonly')).toHaveStyleRule(
      'position', 'absolute',
      'width', '1px',
      'height', '1px',
      'padding', '0',
      'margin', '-1px',
      'overflow', 'hidden',
      'clip', 'rect(0, 0, 0, 0)',
      'white-space', 'nowrap',
      'border-width', '0',
    );
  });
});
