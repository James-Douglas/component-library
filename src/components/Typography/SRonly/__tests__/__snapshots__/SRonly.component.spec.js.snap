// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SRonly.component renders correctly with minimal props 1`] = `
.c0 {
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  white-space: nowrap;
  border-width: 0;
}

@media (min-width:1200px) {

}

@media (min-width:1200px) {

}

@media (min-width:1200px) {

}

@media (min-width:1200px) {

}

@media (min-width:1200px) {

}

@media (min-width:1200px) {

}

<div>
  <div
    class="c0"
  />
</div>
`;
